array-manipulation-1/src/compact.js

```diff
@@ -0,0 +1,13 @@
+/* exported compact */
+function compact(array) {
+  var newArray = [];
+  for (var i = 0; i < array.length; i++) {
+    if (Boolean(array[i]) === true) {
+      newArray.push(array[i]);
+    } else {
+      delete array[i];
+    }
```

Comment:
you can leave this part out. in general we try to avoid messing with the inputs by deleting things from them.